index.html

```diff
@@ -238,13 +238,18 @@ <h3>Introduction</h3>
 <section>
 <h3>Resources Included in the <a>PerformanceResourceTiming</a>
 Interface</h3>
-<p>All resources <a data-cite="FETCH#concept-fetch">fetched</a> by
-the current <a data-cite="HTML#browsing-context">browsing</a> or
-worker [[WORKERS]] context's MUST be included as
+<p>All resource <a data-cite="Fetch#concept-request">Request</a>s
+<a data-cite="FETCH#concept-fetch">fetched</a> by the current
+<a data-cite="HTML#browsing-context">browsing context</a>'s
+<a data-cite="HTML#active-document">active document</a>'s
+<a data-cite="HTML#relevant-settings-object">relevent settings object</a>
```

Comment:
You dropped the bit about workers. It's also rather hard to review this as it's unclear what the context for "current" is. The user agent? If you're already in some scope, this seems wrong.

Reply:
What would be a better way to define "current"? You're right about workers. How can I better refer them here?

Reply:
Where in the spec is this used? If this is some API it ideally would return some pre-existing collection that other specifications produce.

Reply:
So, maybe something like """
* Resource requests which client is not null and which destination is not "document", "object" or "embed" should be added to the request client's relevant global object's performance timeline.
* Resource requests which destination is "document",  which were triggered by the "process the iframe attribute" step, and which client has a document which has a browsing context which has a parent browsing context, should be added to the request's client's document's browsing context's parent browsing context's relevant global object's performance timeline
"""

There's probably a better way to phrase this, but let me know if that would properly define what we need.

Reply:
Things with destinations of "object" or "embed" should always be added, no?

I'm not sure I understand why we need all the conditions listed in the "document" case.  But going from request client to browsing context to parent browsing context looks wrong to me, per the discussion above: the client is already the settings object of the parent.

A settings object has no "relevant global".  I'm not sure what the best way in spec terms is to go from a settings object to the correct performance timeline object.  @domenic, do you know?

Reply:
https://html.spec.whatwg.org/#concept-settings-object-global (if they're indeed associated with global objects).

Reply:
A performance timeline is (kinda) associated with a global object: https://w3c.github.io/performance-timeline/#performance-timeline

At the same time, it also seemed to be owned by an ES global environment. That's probably a mistake and it should be owned by https://html.spec.whatwg.org/multipage/webappapis.html#concept-settings-object-global

Reply:
OK, not sure how, but missed [the conclusion  from the other thread](https://github.com/w3c/resource-timing/pull/177#discussion_r230745051) when writing my previous comment here.

In that light, would a simple "resource Requests which client is not null should be added to the request client's Realm's global object's performance timeline" work?


> Things with destinations of "object" or "embed" should always be added, no?

Indeed. Got myself confused.

Reply:
I don't fully understand what's going on here but I do want to say that you should never have to go through all three levels of the 1:1:1 concepts to get to each other. So

> request client's Realm's global object's

can just be

> request client's global object's

See https://html.spec.whatwg.org/multipage/webappapis.html#realms-settings-objects-global-objects whose second paragraph lists the definitions for navigating from each of realm/settings object/global object to each other. (In this case you'd link "global object" to https://html.spec.whatwg.org/multipage/webappapis.html#concept-settings-object-global .)

Reply:
> would a simple "resource Requests which client is not null should be added to the request client's Realm's global object's performance timeline" work

No, because for the "document" case you want to censor loads not triggered by https://html.spec.whatwg.org/multipage/obsolete.html#process-the-frame-attributes or https://html.spec.whatwg.org/multipage/iframe-embed-object.html#process-the-iframe-attributes afaict.

Reply:
I may very well be wrong, but isn't the "incumbent settings object" in [location object navigate](https://html.spec.whatwg.org/multipage/history.html#location-object-navigate) the settings object of the iframe's browsing context (rather than its parent)?
Similarly in step 13 of [shared-declarative-refresh-steps](https://html.spec.whatwg.org/#shared-declarative-refresh-steps), isn't the source browsing context the iframe's document browsing context?

Reply:
> isn't the "incumbent settings object" in location object navigate the settings object of the iframe's browsing context (rather than its parent)?

The incumbent settings object is the place where the `href` set or `assign` call happens.  It could be anything at all, including the iframe itself, its parent, or some other window  that has its hands on the Location object.

> Similarly in step 13 of shared-declarative-refresh-steps, isn't the source browsing context the iframe's document browsing context?

Yes it is.  I'm not sure why that matters here...   we want to exclude the loads done by https://html.spec.whatwg.org/#shared-declarative-refresh-steps from resource timing, right?

Reply:
> > isn't the "incumbent settings object" in location object navigate the settings object of the iframe's browsing context (rather than its parent)?
> 
> The incumbent settings object is the place where the `href` set or `assign` call happens. It could be anything at all, including the iframe itself, its parent, or some other window that has its hands on the Location object.

OK, thanks for correcting.

> > Similarly in step 13 of shared-declarative-refresh-steps, isn't the source browsing context the iframe's document browsing context?
> 
> Yes it is. I'm not sure why that matters here... we want to exclude the loads done by https://html.spec.whatwg.org/#shared-declarative-refresh-steps from resource timing, right?

We want to exclude it from the parent's resource timing, so for this case it won't be added to it right?

Regardless, I'll add the conditions to only include "document" destinations triggered by "process the iframe-attribute"

Reply:
> We want to exclude it from the parent's resource timing

Right.  That would be handled if we only add navigations that are caused by processing frame or iframe attributes, afaict...

Reply:
aside: should the same exclusions be added to prevent Service Workers from observing the same URLs?

Reply:
I haven't been following the service worker stuff closely, but my understanding is that generally navigations _do_ want to be seen by service workers, but how one determines which service worker should see the navigation is not entirely clear to me and is not really related to  resource timing.

For example, in the case of  `<iframe src="some-cross-origin-url">` we want the cross-origin url to show up in resource timing on the parent page of the iframe but the service worker that  gets used should be the one for  the target origin, not the one for the page the iframe is in, right?